azbfs/datalake_storage_transforms.go

```diff
@@ -0,0 +1,31 @@
+package azbfs
+
+import (
+	"encoding/base64"
+	"strings"
+)
+
+// Converts metadata into a string of format "key1=value1, key2=value2" and Base64 encodes the values.
+func BuildMetadataString(md map[string]string) *string {
```

Comment:
Not sure what the best path is for how to handle metadata. It looks like it was autogenerated in blobs, should I handwrite something similar here?

Reply:
Whatever was auto generated for block-blob, can't same be used here if they are in some common file. If its there why shall we create duplicate code.

Reply:
azcopy seems to have different types for different services, so I dont think we should mix azblob namespaces in the azbfs code. Maybe someone on Ze's team can clarify too.

Reply:
Unfortunately our SDKs do contain duplicate types (e.g. SharedKeyCredential) for the sake of preventing breaking each other with each release. That is a more historical decision, from before the days of `go mod`.

I think with the more modern go.mod, we _could_ definitely do for a azure-storage-core-go instead in our track 2 SDKs. @zezha-msft what do you think?